scan-payment/src/main/java/com/getbouncer/scan/payment/analyzer/NameDetectAnalyzer.kt

```diff
@@ -19,46 +26,71 @@ private const val NUM_PREDICTION_STRIDES = 10
 private const val NMS_THRESHOLD = 0.85F
 private const val CHAR_CONFIDENCE_THRESHOLD = 0.6
 
-class NameDetectAnalyzer private constructor(
-    private val ssdObjectDetect: SSDObjectDetect?,
-    private val alphabetDetect: AlphabetDetect?
-) : Analyzer<SSDOcr.Input, PaymentCardOcrState, NameDetectAnalyzer.Output> {
+class NameAndExpiryAnalyzer private constructor(
+    private val textDetector: TextDetector?,
+    private val alphabetDetect: AlphabetDetect?,
+    private val expiryDetect: ExpiryDetect? = null
+) : Analyzer<SSDOcr.Input, PaymentCardOcrState, NameAndExpiryAnalyzer.Output> {
 
     data class Output(
         val name: String?,
-        val boxes: List<DetectionBox>?
+        val boxes: List<DetectionBox>?,
+        val expiry: ExpiryDetect.Expiry?
     )
 
-    fun isAvailable() = ssdObjectDetect != null
+    fun isAvailable() = textDetector != null
 
     override val name: String = "name_detect_analyzer"
 
     override suspend fun analyze(
         data: SSDOcr.Input,
         state: PaymentCardOcrState
-    ) = if (!state.runNameExtraction || ssdObjectDetect == null || alphabetDetect == null) {
-        Output("", null)
+    ) = if ((!state.runNameExtraction && !state.runExpiryExtraction) || textDetector == null || alphabetDetect == null) {
+        Output(null, null, null)
     } else {
-        val objDetectInput = SSDObjectDetect.Input(
-            fullImage = data.fullImage,
-            previewSize = data.previewSize,
-            cardFinder = data.cardFinder,
-            iin = null
+        val objBitmap = SSDObjectDetect.cropImage(
+            SSDObjectDetect.Input(
+                data.fullImage,
+                data.previewSize,
+                data.cardFinder,
+                null
+            )
         )
 
-        val output = ssdObjectDetect.analyze(objDetectInput, Unit)
-        var nameRect: RectF? = null
+        val future = textDetector.analyze(TextDetector.Input(objBitmap), Unit)
```

Comment:
Can we move the image cropping into the `TextDetector`? I would prefer to keep the image manipulation and logic all in the same place.

Reply:
Why is this named `future`? It looks like it's an instance of `TextDetector.Prediction`.